ocs_ci/deployment/deployment.py

```diff
@@ -193,31 +189,41 @@ def get_olm_and_subscription_manifest():
                     'default_latest_tag', 'latest'
                 )
             )
-        olm_data_generator = templating.load_yaml(
-            constants.OLM_YAML, multi_document=True
+        catalog_source_data = templating.load_yaml(
+            constants.CATALOG_SOURCE_YAML
         )
-        olm_yaml_data = []
         cs_name = constants.OPERATOR_CATALOG_SOURCE_NAME
         # TODO: Once needed we can also set the channel for the subscription
         # from config.DEPLOYMENT.get('ocs_csv_channel')
-        for yaml_doc in olm_data_generator:
-            change_cs_condition = (
-                (image or image_tag) and yaml_doc['kind'] == 'CatalogSource'
-                and yaml_doc['metadata']['name'] == cs_name
+        change_cs_condition = (
+            (image or image_tag) and catalog_source_data['kind'] == 'CatalogSource'
+            and catalog_source_data['metadata']['name'] == cs_name
+        )
+        if change_cs_condition:
+            default_image = config.DEPLOYMENT['default_ocs_registry_image']
+            image = image if image else default_image.split(':')[0]
+            catalog_source_data['spec']['image'] = (
+                f"{image}:{image_tag if image_tag else 'latest'}"
             )
-            if change_cs_condition:
-                default_image = config.DEPLOYMENT['default_ocs_registry_image']
-                image = image if image else default_image.split(':')[0]
-                yaml_doc['spec']['image'] = (
-                    f"{image}:{image_tag if image_tag else 'latest'}"
-                )
-            olm_yaml_data.append(yaml_doc)
-        olm_manifest = tempfile.NamedTemporaryFile(
-            mode='w+', prefix='olm_manifest', delete=False
+        catalog_source_manifest = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='catalog_source_manifest', delete=False
         )
         templating.dump_data_to_temp_yaml(
-            olm_yaml_data, olm_manifest.name
+            catalog_source_data, catalog_source_manifest.name
         )
+        run_cmd(f"oc create -f {catalog_source_manifest.name}", timeout=2400)
+        catalog_source = CatalogSource(
+            resource_name='ocs-catalogsource',
```

Comment:
do you want to take resource_name from constants.OPERATOR_CATALOG_SOURCE_NAME

Reply:
done